## Assignment 3: Finish z-test lab
###### Author: Daniel Fay

In [9]:
# Grab time data from URL
!curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   1732      0 --:--:-- --:--:-- --:--:--  1731


In [2]:
# Import text file into pandas dataframe and print head
import pandas as pd
df = pd.read_csv('times.txt', sep=" ", header = None)
df.columns = ['times']
df.head()

times
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550


##### A new bus route (X8) has been implemented by the MTA. They want to know if it improves commute time?

Null Hypothesis (H0): The commute time is the same or longer with the implementation of the new bus route X8.  
(H0: U1 >= U2)

Alternative Hypthesis (Ha): The commute time is shorter with the implementation of the new bus route X8.  
(HA: U1 < U2)

Significance Level: alpha = 0.05  
Before Implementation: mean = 36, std =6

In [11]:
# Calculate new sample mean and Z score
from __future__ import division
import math
old_mean = 36
old_std = 6
new_mean = df['times'].mean()
z_score = (new_mean - old_mean)/(old_std/(math.sqrt(len(df))))
print 'Z Score = ',z_score

Z Score =  -2.55639718617


This is a left sided test therefore the Z-score which corresponds to an alpha = .05 is -1.645.  
Since -2.5564 is less than -1.645 we reject the null hypothesis and accept the alternative hypothesis.